crates/biome_js_analyze/src/semantic_analyzers/correctness/use_hook_at_top_level.rs

```diff
@@ -292,55 +326,73 @@ impl Rule for UseHookAtTopLevel {
     type Query = FunctionCall;
     type State = Suggestion;
     type Signals = Option<Self::State>;
-    type Options = HooksOptions;
+    type Options = ();
```

Comment:
This is a breaking change @arendjr , we can't remove options from from one minor to another.

My suggestion is to keep the options and **don't read them**. Then you'll have to update the deserialization of the options and add a diagnostic for deprecated options.

Reply:
We could also add a warning telling that the option does nothing?

Reply:
The options are still used for the `useExhaustiveDependencies` rule, so they're not gone. My understanding is that wouldn't invalidate any configs, or am I mistaken?

But indeed, if necessary I guess the best way to deal with it is a deprecation notice saying the options aren't used for this rule anymore.

Reply:
Oh wait, I see. They can indeed be passed to a rule-specific `options` object, which indeed would no longer be valid then. I'll re-add the dependency and indeed print a deprecation warning when anything is set for this rule.

Reply:
I have added a `DeprecatedHooksOptions` type for this linter that accepts the `hooks` array from `HooksOptions`, but prints a deprecation warning when it encounters this.

I have also added a test case for this. Unfortunately it doesn't capture the diagnostic, but at least it shows the old config is still accepted.